From 5daedd627358a1e7a93126a1ecc1d0f336933bd4 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Fri, 15 Aug 2025 10:27:54 +0800
Subject: [PATCH 57/57] 10_linux: pagesize -> configuration

Since we may have other variant specific configurations other than page
sizes. This makes it easier for other maintainers to add localized
strings.
---
 util/grub.d/10_linux.in | 30 +++++++++++++++++++-----------
 1 file changed, 19 insertions(+), 11 deletions(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 98d1d828b..a9d858969 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -98,7 +98,7 @@ done
 # Returns the default page size suffix.
 # WARN $arch is the output of `uname -m`, which is not exactly the same as
 # `dpkg --print-architecture`.
-default_pagesize() {
+default_configuration() {
 	case "$arch" in
 		loongarch64)
 			printf "16k"
@@ -110,24 +110,24 @@ default_pagesize() {
 }
 
 # Localize page size comment string.
-pagesize_str() {
+configuration_str() {
 	case "$1" in
 		4k)
-			if [ "$(default_pagesize)" = "$1" ] ; then
+			if [ "$(default_configuration)" = "$1" ] ; then
 				printf ""
 			else
 				gettext_printf "4K Page Size"
 			fi
 			;;
 		16k)
-			if [ "$(default_pagesize)" = "$1" ] ; then
+			if [ "$(default_configuration)" = "$1" ] ; then
 				printf ""
 			else
 				gettext_printf "16K Page Size"
 			fi
 			;;
 		64k)
-			if [ "$(default_pagesize)" = "$1" ] ; then
+			if [ "$(default_configuration)" = "$1" ] ; then
 				printf ""
 			else
 				gettext_printf "64K Page Size"
@@ -159,13 +159,21 @@ variant_str() {
 }
 
 # Break localversion apart, return the processed comments.
+# The LOCALVERSION comes with several parts:
+#   -VENDOR-VARIANT-CONFIGURATOIN
+# Where:
+#   - VENDOR: Undoubtably, "aosc".
+#   - VARIANT: The variant of the kernel, built for specific purposes or
+#     devices, e.g. "main" for mainline kernel, "lts" for LTS kernel, and
+#     "asahi" for Apple Silicon devices.
+#   - CONFIGURATION: Variant specific configurations, e.g. page sizes.
 # E.g.
 # - "-aosc-main" -> ""
 # - "-aosc-lts" -> " (LTS Kernel)"
 # - "-aosc-main-4k" -> " (4K Page Size)"
 # - "-aosc-rc-4k" -> " (RC Kernel, 4K Page Size)"
 get_comments() {
-	local _str="$1" _fullver="$1" segs=() comments=() sep=", " variant pagesize
+	local _str="$1" _fullver="$1" segs=() comments=() sep=", " variant configuration
 	# Not an AOSC OS kernel
 	if ! [[ "$_str" =~ \.[0-9]+-aosc- ]] ; then
 		return
@@ -176,8 +184,8 @@ get_comments() {
 	segs=($_str)
 	unset IFS
 	variant="$(variant_str "${segs[0]}")"
-	pagesize="$(pagesize_str "${segs[1]}")"
-	comment="${variant}${sep}${pagesize}"
+	configuration="$(configuration_str "${segs[1]}")"
+	comment="${variant}${sep}${configuration}"
 	if [ "x$comment" = "x$sep" ] ; then
 		return
 	fi
@@ -255,10 +263,10 @@ find_rootdev() {
 # The kernel with the default size will be always on top. If it does
 # not exist, then the kernel without page size suffix will be on top.
 prioritize() {
-	local kernels newkernels default_pgsize flag
+	local kernels newkernels default_configuration flag
 	kernels=("$@")
 	newkernels=()
-	default_pgsize="$(default_pagesize)"
+	default_configuration="$(default_configuration)"
 	for k in "${kernels[@]}" ; do
 		# This system probably don't have other page sizes.
 		if [ "${k%%-[0-9]*k}" = "$k" ] ; then
@@ -269,7 +277,7 @@ prioritize() {
 			fi
 			continue
 		fi
-		if [ "${k%%-$default_pgsize}" != "$k" ] ; then
+		if [ "${k%%-$default_configuration}" != "$k" ] ; then
 			newkernels=("$k" "${newkernels[@]}")
 			continue
 		fi
-- 
2.50.1

